modelforge/tests/test_training.py

```diff
@@ -80,36 +75,100 @@ def get_trainer(potential_name: str, dataset_name: str, training_toml: str):
+    t_config.loss_parameter.weight["per_molecule_dipole_moment"] = 0.01
+    t_config.loss_parameter.weight["per_molecule_total_charge"] = 0.01
+
+    # also add per_atom_charge to predicted properties
+
+    p_config = config["potential"]
+    p_config.core_parameter.predicted_properties.append("per_atom_charge")
+    p_config.core_parameter.predicted_dim.append(1)
+
+
+def replace_per_molecule_with_per_atom_loss(config):
+    t_config = config["training"]
+    t_config.loss_parameter.loss_property.remove("per_molecule_energy")
+    t_config.loss_parameter.loss_property.append("per_atom_energy")
+
+    t_config.loss_parameter.weight.pop("per_molecule_energy")
+    t_config.loss_parameter.weight["per_atom_energy"] = 0.999
+
+    # NOTE: the loss is calculate per_atom, but the validation set error is
+    # per_molecule. This is because otherwise it's difficult to compare.
+    t_config.early_stopping.monitor = "val/per_molecule_energy/rmse"
+    t_config.monitor_for_checkpoint = "val/per_molecule_energy/rmse"
+    t_config.lr_scheduler.monitor = "val/per_molecule_energy/rmse"
+
+
 @pytest.mark.skipif(ON_MACOS, reason="Skipping this test on MacOS GitHub Actions")
 @pytest.mark.parametrize(
     "potential_name", _Implemented_NNPs.get_all_neural_network_names()
 )
 @pytest.mark.parametrize("dataset_name", ["PHALKETHOH"])
-@pytest.mark.parametrize("training", ["with_force", "without_force"])
-def test_train_with_lightning(training, potential_name, dataset_name):
+@pytest.mark.parametrize(
+    "loss",
+    ["energy", "energy_force", "normalized_energy_force", "energy_force_dipole_moment"],
+)
+def test_train_with_lightning(loss, potential_name, dataset_name):
     """
     Test that we can train, save and load checkpoints.
     """
-    # get correct training toml
-    training_toml = "default_with_force" if training == "with_force" else "default"
 
     # SKIP if potential is ANI and dataset is SPICE2
     if "ANI" in potential_name and dataset_name == "SPICE2":
         pytest.skip("ANI potential is not compatible with SPICE2 dataset")
-    if IN_GITHUB_ACTIONS and potential_name == "SAKE" and training == "with_force":
+    if IN_GITHUB_ACTIONS and potential_name == "SAKE" and "force" in loss:
         pytest.skip(
             "Skipping Sake training with forces on GitHub Actions because it allocates too much memory"
         )
 
-    # train potential
-    get_trainer(
-        potential_name, dataset_name, training_toml
-    ).train_potential().save_checkpoint(
-        "test.chp"
-    )  # save checkpoint
+    config = load_configs_into_pydantic_models(potential_name, dataset_name)
 
+    if "force" in loss:
+        add_force_to_loss_parameter(config)
+    if "normalized" in loss:
+        replace_per_molecule_with_per_atom_loss(config)
+    if "dipole_moment" in loss:
+        add_dipole_moment_to_loss_parameter(config)
+
+    # train potential
+    get_trainer(config).train_potential().save_checkpoint("test.chp")  # save checkpoint
```

Comment:
the `get_trainer` function might be an idea instead of the factory class @MarshallYan @chrisiacovella